pkgs/development/tools/gnulib/default.nix

```diff
@@ -26,6 +26,17 @@ stdenv.mkDerivation {
   # do not change headers to not update all vendored build files
   dontFixup = true;
 
+  passthru = {
+    # This patch is used by multiple other packages (currently:
+    # gnused, gettext) which contain vendored copies of gnulib.
+    # Without it, compilation will fail with error messages about
+    # "__LDBL_REDIR1_DECL" or similar on platforms with longdouble
+    # redirects (currently powerpc64).  Once all of those other
+    # packages make a release with a newer gnulib we can drop this
+    # patch.
```

Comment:
Can't we add gnulib to buildInputs and run autoreconf?

Reply:
Unfortunately no.  Gnulib's vendoring works similarly to autoconf, but it's a separate system; they have their own tool [`gnulib-tool`](https://www.gnu.org/software/gnulib/manual/html_node/Gnulib-Basics.html) that does something similar to what `autoreconf` does.

I did at one point consider writing a `gnulibRevendorHook` analogous to `autoreconfHook`.

However the situation with gnulib is much hairier: they don't have numbered versions/releases, and there is a very complex API between gnulib and the gnulib-using code.  This API doesn't have version numbers either.  So it's quite a bit worse than the situation with autoconf where we keep a dozen-ish versions around... with gnulib every git commit is potentially a backwards-incompatible ugprade.

Lastly, `gnulib-tool` is just part of the "re-vendoring" process; many projects seem to have project-specific maintainer scripts that need to be used to invoke `gnulib-tool`.  In some cases (`gettext`) I was unable to find the script that upstream uses to do the vendoring and make a release tarball.

My impression is that gnulib is sort of a relic left over from the days when the vast majority of UNIXes were closed-source.  All uses of gnulib I'm aware of were introduced to their host package more than 20 years ago.